src/zabapgit_git.prog.abap

```diff
@@ -1043,7 +1043,18 @@ CLASS lcl_git_pack IMPLEMENTATION.
                                               iv_length = 4 ).
     CONCATENATE rv_data lv_xstring INTO rv_data IN BYTE MODE.
 
+    lv_objects_total = lines( it_objects ).
+
     LOOP AT it_objects ASSIGNING <ls_object>.
+
+      lv_objects_processed = sy-tabix.
+
+      cl_progress_indicator=>progress_indicate(
+        EXPORTING
+          i_text               = |encoding objects &1% ( &2 of &3 )|
+          i_processed          = lv_objects_processed
+          i_total              = lv_objects_total ).
```

Comment:
FM SAPGUI_PROGRESS_INDICATOR is called only every 10 seconds - this should be ok...